lib/channel.js

```diff
@@ -98,21 +128,71 @@ class Channel {
     const packet = metautil.jsonParse(data);
     if (!packet) {
       const error = new Error('JSON parsing error');
-      this.error(400, { error, pass: true });
+      this.error(500, { error, pass: true });
       return;
     }
+    const [callType] = Object.keys(packet);
+    if (callType === 'call') return this.handleRpcPacket(packet);
+    else if (callType === 'stream') return this.handleStreamPacket(packet);
+    const error = new Error('Packet structure error');
+    this.error(500, { error, pass: true });
+  }
+
+  binary(data) {
+    try {
+      const { streamId, payload } = MetacomChunk.decode(data);
+      const upstream = this.client.streams.get(streamId);
+      if (upstream) {
+        upstream.push(payload);
+      } else {
+        const error = new Error(`Stream ${streamId} is not initialized`);
+        this.error(400, { callId: streamId, error, pass: true });
+      }
+    } catch (error) {
+      this.error(400, { callId: 0, error });
+    }
+  }
+
+  handleRpcPacket(packet) {
     const [callType, target] = Object.keys(packet);
     const callId = parseInt(packet[callType], 10);
     const args = packet[target];
     if (callId && args) {
       const [interfaceName, methodName] = target.split('/');
-      this.rpc(callId, interfaceName, methodName, args);
+      void this.rpc(callId, interfaceName, methodName, args);
```

Comment:
Why do we need void here?

Reply:
It's a good practice to dispose the resulting promise with `void`, when we don't want to `await` async functions. It also explicitly makes clear that we **intentionally** do not want to unpack this promise. 
A simple marker for not awaiting, which can be actually evaluated as a mistake by other developers.